scheds/rust/scx_p2dq/src/bpf/main.bpf.c

```diff
@@ -420,20 +420,20 @@ static s32 pick_idle_cpu(struct task_struct *p, struct task_ctx *taskc,
 	 */
 	if (wake_flags & SCX_WAKE_SYNC) {
 		struct task_struct *current = (void *)bpf_get_current_task_btf();
-		if (!current)
-			goto out_put_cpumask;
-
 		struct task_ctx *cur_taskc = lookup_task_ctx_may_fail(current);
-		if (!cur_taskc)
-			goto out_put_cpumask;
+		if (cur_taskc) {
+			// XXX: for some reason this check never passes
+			if (scx_bpf_test_and_clear_cpu_idle(cur_taskc->cpu)) {
+				stat_inc(P2DQ_STAT_LOCAL_WAKEUP);
```

Comment:
I tried a bunch of different workloads but this path never seems to be hit and the waker CPU never seems to be idle.

Reply:
Yeah, now that I think about this again that does make sense. Let me think about how to better handle this.